tests/dtm/test_rgw_s3_process_restart.py

```diff
@@ -841,10 +857,105 @@ def test_bkt_creation_ios_after_rgw_restart(self):
         self.log.info("Step 3: Perform WRITEs/READs-Verify/DELETEs with variable sizes objects.")
         resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=self.iam_user,
                                                     log_prefix=test_prefix,
-                                                    nclients=test_cfg['nclients'],
-                                                    nsamples=test_cfg['nsamples'])
+                                                    nclients=self.test_cfg['clients'],
+                                                    nsamples=self.test_cfg['test_42256'][
+                                                        'nsamples'])
         assert_utils.assert_true(resp[0], resp[1])
         self.log.info("Step 3: Successfully performed WRITEs/READs-Verify/DELETEs with variable "
                       "sizes objects.")
 
         self.log.info("ENDED: Verify bucket creation and IOs after rgw restart using pkill")
+
+    @pytest.mark.lc
+    @pytest.mark.dtm
+    @pytest.mark.tags("TEST-42257")
+    def test_mpu_rgw_restart(self):
+        """Verify multipart upload and download before/after rgw is restarted."""
+        self.log.info("STARTED: Verify multipart upload and download before/after rgw is restarted")
+
+        file_size = HA_CFG["5gb_mpu_data"]["file_size"]
+        total_parts = HA_CFG["5gb_mpu_data"]["total_parts"]
+        part_numbers = list(range(1, total_parts + 1))
+        self.system_random.shuffle(part_numbers)
+        multipart_obj_path = os.path.join(self.test_dir_path, "test_42257_file")
+        download_path = os.path.join(self.test_dir_path, "test_42257_file_download")
+        part_numbers = list(range(1, total_parts + 1))
+        self.system_random.shuffle(part_numbers)
+        output = Queue()
+        parts_etag = list()
+        event = threading.Event()  # Event to be used to send intimation of cluster restart
+
+        self.log.info("Step 1: Start multipart upload of 5GB object in background")
+        args = {'s3_data': self.iam_user, 'bucket_name': self.bucket_name,
+                'object_name': self.object_name, 'file_size': file_size, 'total_parts': total_parts,
+                'multipart_obj_path': multipart_obj_path, 'part_numbers': part_numbers,
+                'parts_etag': parts_etag, 'output': output}
+        thread = threading.Thread(target=self.ha_obj.start_random_mpu, args=(event,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        self.log.info("Step 1: Started multipart upload of 5GB object in background")
+
+        time.sleep(self.delay)
```

Comment:
This will need more delay as start_random_mpu method creates bucket, creates files and creates data parts and then starts upload.